File: backend/app/schemas/risk_notice_card.py
from typing import Literal
from pydantic import BaseModel


class SignItem(BaseModel):
    category: Literal["warning", "prohibition", "instruction", "notice"]
    name: str
    svg_name: str


class RightColumn(BaseModel):
    hazard_description: str = ""
    accident_types: list[str] = []
    control_measures: list[str] = []
    emergency_measures: list[str] = []
    signs: list[SignItem] = []
    signs_source: str | None = None


class CardData(RightColumn):
    object_id: str
    enterprise_name: str
    name: str
    code: str
    level: str
    inherent_risk_level: str | None = None
    level_color: str
    responsible_unit: str
    responsible_person: str
    contact_phone: str
    fallback_used: bool = False
    has_open_hazard: bool = False
    signs: list[SignItem] = []
    snapshot: dict | None = None
    stale: bool = False
    public_url: str
    generated_at: str


class CardSummary(BaseModel):
    object_id: str
    name: str
    zone_name: str = ""
    level: str
    level_color: str
    accident_types: list[str] = []
    signs: list[SignItem] = []
    responsible_unit: str = ""
    has_open_hazard: bool = False
    snapshot: dict | None = None
    stale: bool = False
    public_url: str


class ExportRequest(BaseModel):
    object_ids: list[str]


class ExportResponse(BaseModel):
    file_key: str
    warnings: list[str] = []


class AiOptimizeResponse(BaseModel):
    original: RightColumn
    optimized: RightColumn


class SignSuggestion(BaseModel):
    remove: list[str] = []
    add: list[str] = []
    reasons: list[dict] = []


class AiSignReviewResponse(BaseModel):
    original_signs: list[SignItem] = []
    suggestion: SignSuggestion
    # 候选库（端点已组装的去重全量标志，供前端人工微调/中文名映射复用）
    catalog: list[SignItem] = []


class SnapshotSaveRequest(BaseModel):
    content: RightColumn


class SnapshotResponse(BaseModel):
    version: int
    source: str


class TokenResetResponse(BaseModel):
    public_url: str

File: frontend/src/types/riskNoticeCard.ts
/** 风险告知卡（Risk Notice Card）类型定义。
 *
 * 结构与后端 backend/app/schemas/risk_notice_card.py 一一对应：
 * SignItem / RightColumn / CardData / CardSummary。
 */

export type SignCategory = "warning" | "prohibition" | "instruction" | "notice";

export interface SignItem {
  category: SignCategory;
  name: string;
  svg_name: string;
}

/** AI 标志审查建议（remove/add 为 svg_name 集合，reasons 为逐项理由）。 */
export interface SignSuggestion {
  remove: string[];
  add: string[];
  reasons: { sign_name: string; reason: string }[];
}

/** AI 标志审查结果（original_signs 为审查时当前标志）。 */
export interface AiSignReviewResponse {
  original_signs: SignItem[];
  suggestion: SignSuggestion;
}

/** 右栏四块内容（AI 优化仅针对 hazard_description/control_measures/emergency_measures）。 */
export interface RightColumn {
  hazard_description: string;
  accident_types: string[];
  control_measures: string[];
  emergency_measures: string[];
}

/** 快照信息（version 每次保存 +1，source 区分 ai/rule）。 */
export interface SnapshotInfo {
  version: number;
  source: string;
}

/** 单卡完整数据（快照优先）。 */
export interface CardData extends RightColumn {
  object_id: string;
  enterprise_name: string;
  name: string;
  code: string;
  level: string;
  level_color: string;
  responsible_unit: string;
  responsible_person: string;
  contact_phone: string;
  fallback_used: boolean;
  signs: SignItem[];
  /** 标志来源（后端回填，旧响应可能缺失）。 */
  signs_source?: "rule" | "ai" | "manual";
  snapshot: SnapshotInfo | null;
  stale: boolean;
  public_url: string;
  generated_at: string;
}

/** 列表摘要（列表端点返回）。 */
export interface CardSummary {
  object_id: string;
  name: string;
  zone_name: string;
  level: string;
  level_color: string;
  accident_types: string[];
  signs: SignItem[];
  responsible_unit: string;
  snapshot: SnapshotInfo | null;
  stale: boolean;
  public_url: string;
}
